### PR TITLE
fix: fix cascader type

### PR DESCRIPTION
**Pull request by crazyair**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix the issue of redundant generics in Cascader      |
| 🇨🇳 Chinese |     修复 Cascader 多余泛型问题     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
原始问题是啥

**Comment by crazyair** [comment]
> 原始问题是啥

如果吧 multiple 类型放到泛型上，如下这种使用，无法自动切换是否 multiple，rc-cascader 写3个泛型是因为，checkable 支持 ReactNode，而 antd 的 Cascader 的 multiple 只是 boolean 类型，所以可以精简泛型

```tsx
 const demo: CascaderType = {
      multiple: true,
      onChange: (value) => {},
    };
    expect(demo).toBeTruthy();
    const demo2: CascaderType = {
      onChange: (value) => {},
    };
    expect(demo2).toBeTruthy();

```

**Comment by crazyair** [comment]
image

加上非必填就不行了。。。

File: components/cascader/index.tsx [hunk 1]
```diff
@@ -135,6 +135,19 @@ export interface CascaderProps<
    */
   variant?: Variant;
 }
+export type CascaderType<
+  OptionType extends DefaultOptionType = DefaultOptionType,
+  ValueField extends keyof OptionType = keyof OptionType,
+> =
+  | CascaderProps<OptionType, ValueField>
+  | (CascaderProps<OptionType, ValueField, true> & { multiple: true });
```
Comment (afc163) [inline components/cascader/index.tsx:143]: CascaderType 这个命名比较奇怪，好像其他组件没有这个 XxxType。

Comment (crazyair) [reply components/cascader/index.tsx:143]: CascaderProps 还得用，所以单加了个 CascaderType

Comment (afc163) [reply components/cascader/index.tsx:143]: 搞一个 `CascaderProps<Option, 'name', 'auto'>`?

Comment (crazyair) [reply components/cascader/index.tsx:143]: 怎么说？

Comment (afc163) [reply components/cascader/index.tsx:143]: 先改成 CascaderAutoProps 吧

Comment (crazyair) [reply components/cascader/index.tsx:143]: done

**Review by afc163 — APPROVED** [review]

**Pull request merged**